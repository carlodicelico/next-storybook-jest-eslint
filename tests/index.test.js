/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import sinon from 'sinon'

import Index from '../pages/index.js'
import expectExport from 'expect'

const showsMock = [
    { id: 1, name: "Foo" },
    { id: 2, name: "Bar" },
    { id: 3, name: "Baz" },
    { id: 4, name: "Qux" }
  ]

describe('<Index />', () => {
  it('shows "Batman TV Shows"', () => {
    const wrapper = mount(<Index shows={ showsMock } />)
    expect(wrapper.find('Index').find('h1').text()).toEqual('Batman TV Shows')
  })
})

describe('<Index /> snapshot', () => {
  it('matches snapshot', () => {
    const component = renderer.create(<Index shows={ showsMock } />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
