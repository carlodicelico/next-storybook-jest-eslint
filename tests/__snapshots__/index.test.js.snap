// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> snapshot matches snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid #DDD",
      "margin": 20,
      "padding": 20,
    }
  }
>
  <div>
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      Home
    </a>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={
        Object {
          "marginRight": 15,
        }
      }
    >
      About
    </a>
  </div>
  <div>
    <h1>
      Batman TV Shows
    </h1>
    <ul>
      <li>
        <a
          href="/p/1"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Foo
        </a>
      </li>
      <li>
        <a
          href="/p/2"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Bar
        </a>
      </li>
      <li>
        <a
          href="/p/3"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Baz
        </a>
      </li>
      <li>
        <a
          href="/p/4"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Qux
        </a>
      </li>
    </ul>
  </div>
</div>
`;
